File: web/src/utils/R.ts
export class R<T = any> {
    code?: number;
    message?: string | undefined;
    data?: T | undefined;

    public get success(): boolean {
        return this.code == 0;
    }

    public set success(val: boolean) {
        this.code = val ? 0 : 1;
    }

    public static ok<T>(data?: T): R<T> {
        const r = new R<T>();
        r.success = true
        r.data = data;
        return r;
    }

    public static error<T>(message: string | undefined, code = 1): R<T> {
        const r = new R<T>();
        r.success = false
        r.code = code;
        r.message = message;
        return r;
    }
}

File: net/XcLive.Client/Data/R.cs
namespace Frame.Data;

public class R
{
    public bool success
    {
        get { return code == 0; }
        set { code = value ? 0 : 1; }
    }

    public int code { get; set; }
    public string message { get; set; }

    public static R OK()
    {
        return new R()
        {
            success = true,
        };
    }

    public static R<T> OK<T>(T data)
    {
        return new R<T>()
        {
            success = true,
            data = data
        };
    }

    public static R<T> Faild<T>(string message)
    {
        return new R<T>()
        {
            message = message,
            success = false,
        };
    }

    public static R Faild(string message)
    {
        return new R()
        {
            success = false, message = message
        };
    }
}

public class R<T> : R
{
    public T data { get; set; }
}